### PR TITLE
[Feature] PaddleOCR-VL: OCR vision-language model on a single Blackhole P150

### PR DESCRIPTION
**Pull request by stisiTT**
### PR Category
Feature

### Summary
Brings up PaddleOCR-VL, a small open-source document-reading AI model, on a single Tenstorrent Blackhole chip (P150). It reads text out of scanned pages and images with no errors on our test set, and runs fast enough to serve live requests through the same API used for other hosted models. Also adds a real automated test suite (previously the checks were print statements only, so nothing caught a regression), and fixes an unrelated startup bug that affected other single-chip models running on shared multi-chip machines.

### Notes for reviewers
- Single P150 (1x1) only for now; no multi-chip support.
- Known limit: images scanned above a certain resolution can make the model repeat text instead of stopping. A resolution ceiling is now enforced and guarded by `tests/test_resolution_ceiling.py`; see the last two commits for the accuracy data behind the chosen limit.
- CI registration (`tests/pipeline_reorg/*.yaml`, tier assignment, staging the test data on shared storage) is intentionally left out of this PR as a follow-up, since the golden test data is too large to commit. Same approach the Qwen3-ASR bring-up PR took this week.
- Code comments in the bring-up commits were trimmed to short one-liners; the design rationale that used to live in those comments now lives in the commit messages instead.
- Includes one small unrelated bug fix (mesh-map entries missing for some Blackhole SKUs), kept in this PR because it was needed to get this model running on this hardware at all.

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:stisi/paddleocr-vl-p150)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:stisi/paddleocr-vl-p150)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:stisi/paddleocr-vl-p150)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:stisi/paddleocr-vl-p150)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-tests.yaml?query=branch:stisi/paddleocr-vl-p150)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:stisi/paddleocr-vl-p150)

### CODE REVIEW

**Comment by yieldthought** [comment]
Codex review:

**[P2] Match the vision fixture's scope to the device fixture.** [models/demos/blackhole/paddleocr_vl/tests/test_vision_tower_pcc.py:39](https://github.com/tenstorrent/tt-metal/blob/2970d1be41080a847bbb662065f0045370386d95/models/demos/blackhole/paddleocr_vl/tests/test_vision_tower_pcc.py#L39) declares a module-scoped `tower(device)`, but [conftest.py:431](https://github.com/tenstorrent/tt-metal/blob/2970d1be41080a847bbb662065f0045370386d95/conftest.py#L431) declares `device` with function scope. When the golden directory is present, pytest raises `ScopeMismatch` before any model inference, so none of the four projector PCC cases can run. A host pytest probe using the exact fixture/test definitions reproduced all four setup errors without opening a device. Suggestion: make `tower` function-scoped; the accompanying inline suggestion does that.

File: models/demos/blackhole/paddleocr_vl/tests/test_vision_tower_pcc.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+from models.demos.blackhole.paddleocr_vl.tests.generate_goldens import INTERMEDIATE_SAMPLES
+from models.demos.blackhole.paddleocr_vl.tt.vision.model import DropInVisionTransformer
+from models.demos.blackhole.paddleocr_vl.tt.vision.vision_model_config import VisionModelArgs
+from models.demos.blackhole.paddleocr_vl.tt.weight_mapping import map_vision_state_dict
+from tests.ttnn.utils_for_testing import assert_with_pcc
+
+PCC_TARGET = 0.98
+
+
+@pytest.fixture(scope="module")
```
Comment (yieldthought) [inline models/demos/blackhole/paddleocr_vl/tests/test_vision_tower_pcc.py]: Codex: `tower` depends on the function-scoped root `device` fixture. Module scope causes `ScopeMismatch` for all four PCC cases once the golden directory is present. Suggestion: use the same function lifetime as the device.

```suggestion
@pytest.fixture
```

Comment (stisiTT) [reply models/demos/blackhole/paddleocr_vl/tests/test_vision_tower_pcc.py]: Fixed in 0c130ad. Dropped `scope="module"` from the `tower` fixture so it matches the function-scoped `device` fixture. One-line change; PCC cases should run through setup now.

**Comment by mtairum** [comment]
@stisiTT do we want to add this model to our model tier CI? https://github.com/tenstorrent/tt-metal/blob/main/models/model_ci_tiers.md?
